### PR TITLE
TUTAND-103 || Fix AppSync new enums issue

### PR DESCRIPTION
**Pull request by admund**
When BE will add a new value for any Enum in the schema and will start sending it the oldest app will crash. I fixed the code generator to handle empty or unknown Enum values.
Fix:
1. I added UNKNOWN enum value to every generated enum.
2. UKNOWN will be set when the enum value is null (to by safe for kotlin) or some new one.

Old generated code:
```     
final MyEnum something;
if (somethingStr != null) {
  something = MyEnum.valueOf(somethingStr);
} else {
  something = null;
}
 ```

New generated code:
```
MyEnum something;
if (somethingStr != null) {
  try {
    something = MyEnum.valueOf(somethingStr);
  } catch (IllegalArgumentException exception) {
    something = MyEnum.UNKNOWN;
  }
} else {
  something = MyEnum.UNKNOWN;
}
```

### CODE REVIEW

File: aws-android-sdk-appsync-compiler/src/main/kotlin/com/amazonaws/apollographql/apollo/compiler/ResponseFieldSpec.kt [hunk 1]
```diff
@@ -99,11 +99,15 @@ class ResponseFieldSpec(
...
+        .addStatement("\$T \$L", normalizedFieldSpec.type, fieldSpec.name)
         .beginControlFlow("if (\$LStr != null)", fieldSpec.name)
+        .beginControlFlow("try")
         .addStatement("\$L = \$T.valueOf(\$LStr)", fieldSpec.name, normalizedFieldSpec.type, fieldSpec.name)
+        .nextControlFlow("catch (IllegalArgumentException exception)")
+        .addStatement("\$L = \$T.UNKNOWN", fieldSpec.name, normalizedFieldSpec.type)
+        .endControlFlow()
         .nextControlFlow("else")
-        .addStatement("\$L = null", fieldSpec.name)
+        .addStatement("\$L = \$T.UNKNOWN", fieldSpec.name, normalizedFieldSpec.type)
```
Comment (Denis-Brainly) [inline aws-android-sdk-appsync-compiler/src/main/kotlin/com/amazonaws/apollographql/apollo/compiler/ResponseFieldSpec.kt:110]: [nit] Sorry for my question, how we will add for each enum that UNKNOWN vale? Could you please share some documentation how this pseudocode works.

**Review by Denis-Brainly — APPROVED** [review]

File: aws-android-sdk-appsync-compiler/src/main/kotlin/com/amazonaws/apollographql/apollo/compiler/ir/TypeDeclaration.kt [hunk 2]
```diff
@@ -58,6 +58,7 @@ data class TypeDeclaration(
             .build()
         addEnumConstant(value.name, typeSpec)
       }
+      addEnumConstant("UNKNOWN")
```
Comment (admund) [inline aws-android-sdk-appsync-compiler/src/main/kotlin/com/amazonaws/apollographql/apollo/compiler/ir/TypeDeclaration.kt:61]: click

File: aws-android-sdk-appsync-compiler/src/main/kotlin/com/amazonaws/apollographql/apollo/compiler/ResponseFieldSpec.kt (hunk 1 above)
Comment (admund) [reply aws-android-sdk-appsync-compiler/src/main/kotlin/com/amazonaws/apollographql/apollo/compiler/ResponseFieldSpec.kt:110]: I'm adding it here: https://github.com/brainly/aws-mobile-appsync-sdk-android/pull/2/files#r749364114

AppSynce uses https://github.com/square/javapoet for generating code, you can look at their documentation.

Comment (scana) [reply aws-android-sdk-appsync-compiler/src/main/kotlin/com/amazonaws/apollographql/apollo/compiler/ResponseFieldSpec.kt:110]: Any chance that this has already been added upstream? 🤔

**Review by scana — APPROVED** [review]

File: aws-android-sdk-appsync-compiler/src/main/kotlin/com/amazonaws/apollographql/apollo/compiler/ResponseFieldSpec.kt (hunk 1 above)
Comment (admund) [reply aws-android-sdk-appsync-compiler/src/main/kotlin/com/amazonaws/apollographql/apollo/compiler/ResponseFieldSpec.kt:110]: I checked newest AppSync lib some time ago and still had this issue. I will create PR to propose this fix in the root repo.

Comment (scana) [reply aws-android-sdk-appsync-compiler/src/main/kotlin/com/amazonaws/apollographql/apollo/compiler/ResponseFieldSpec.kt:110]: Great idea! <3

**Pull request merged**